src/ptracer.rs

```diff
@@ -178,6 +270,24 @@ impl Tracee {
         }
     }
 
+    #[cfg(all(target_os = "android", target_arch = "aarch64"))]
+    pub fn debug_register(&self, regtype: DebugRegisterType, index: usize) -> Result<user_hwdebug_state_reg> {
+        assert!(index < 16);
```

Comment:
Or: return the full `user_hwdebug_state` and leave it to callers to update it appropriately. Since the actual underlying API is just "read the entire debug register context, let the user mess with it, write the entire context", we would typically have `pete` just wrap that way of interacting.

Reply:
I've taken the second option.